src/models/gun.rs

```diff
@@ -0,0 +1,73 @@
+use std::time::{Instant};
+
+const DEFAULT_NATURAL_COOL_DOWN_RATE: f32 = 0.2;
+const DEFAULT_OVERHEAT_COOL_DOWN_RATE: f32 = 0.40;
+const DEFAULT_MAXIMUM_CAPACITY: f32 = 1.0;
+const DEFAULT_MINIMUM_CAPACITY: f32 = 0.0;
+const DEFAULT_HEAT_UP: f32 = 0.05;
+
+pub struct Gun {
+    pub minimum_capacity: f32,
+    pub maximum_capacity: f32,
+    pub temperature: f32,
+    natural_cool_down_rate: f32,
+    overheated: bool,
+    overheat_cool_down_rate: f32,
+    last_update: Instant
+}
+
+
+impl Default for Gun {
+    fn default() -> Gun { Gun::new() }
+}
+
+impl Gun {
+    pub fn new() -> Gun {
+        let maximum_capacity = DEFAULT_MAXIMUM_CAPACITY;
+        let minimum_capacity = DEFAULT_MINIMUM_CAPACITY;
+        Gun {
+            minimum_capacity: minimum_capacity,
+            maximum_capacity: maximum_capacity,
+            temperature: minimum_capacity,
+            natural_cool_down_rate: DEFAULT_NATURAL_COOL_DOWN_RATE * maximum_capacity,
+            overheated: false,
+            overheat_cool_down_rate: DEFAULT_OVERHEAT_COOL_DOWN_RATE * maximum_capacity,
+            last_update: Instant::now()
+        }
+    }
+    pub fn cool_down(&mut self){
```

Comment:
We should pass a `seconds: f32` parameter, that amounts to the number of seconds elapsed since the last time the `cool_down` function was called. That way we don't have to use `Instant::now`.

This would require passing the same parameter down to `overheat_cool_down` and `natural_cool_down`.